synapse/storage/group_server.py

```diff
@@ -29,6 +30,18 @@
 
 
 class GroupServerStore(SQLBaseStore):
+    def set_group_join_policy(self, group_id, join_policy):
+        return self._simple_update_one(
```

Comment:
`set_group_join_policy` needs a docstring, especially as we only want to accept certain values so we should document what they are.

Might also be worth adding a `assert join_policy in ('invite', 'open')` (we should probably move some of those to constants somewhere at some point)